database/entity/0061-event_refactoring/Event.ts

```diff
@@ -0,0 +1,86 @@
+import { Contribution } from '../Contribution'
+import { ContributionMessage } from '../ContributionMessage'
+import { User } from '../User'
+import { Transaction } from '../Transaction'
+import Decimal from 'decimal.js-light'
+import {
+  BaseEntity,
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  CreateDateColumn,
+  ManyToOne,
+  JoinColumn,
+} from 'typeorm'
+import { DecimalTransformer } from '../../src/typeorm/DecimalTransformer'
+
+@Entity('events')
+export class Event extends BaseEntity {
+  @PrimaryGeneratedColumn('increment', { unsigned: true })
+  id: number
+
+  @Column({ length: 100, nullable: false, collation: 'utf8mb4_unicode_ci' })
+  type: string
+
+  @CreateDateColumn({
+    name: 'created_at',
+    type: 'datetime',
+    default: () => 'CURRENT_TIMESTAMP()',
+    nullable: false,
+  })
+  createdAt: Date
+
+  @Column({ name: 'affected_user_id', unsigned: true, nullable: false })
+  affectedUserId: number
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: 'affected_user_id', referencedColumnName: 'id' })
+  affectedUser: User
+
+  // TODO potentially save actingRole as well
```

Comment:
Since we do not have a proper role model yet this is just something to make an issue out of.

If you think about user confirming contributions as you suggested in one of your ideas the same event could be triggered by different roles: `confirmContribution` - one time as user, one time as moderator.
Nothing important for now - its just a thought I had, when I designed the new Event table.